tests/unit/test_contrib/test_prometheus.py

```diff
@@ -1,217 +1,63 @@
-        )
-
-        assert (
-            """litestar_rocks_request_duration_seconds_bucket{app_name="litestar_test",baz="qux",foo="bar",le="0.5",method="GET",path="/test",status_code="200"} 1.0"""
-            in metrics
-        )
-
-        assert (
-            """litestar_rocks_request_duration_seconds_bucket{app_name="litestar_test",baz="qux",foo="bar",le="1.0",method="GET",path="/test",status_code="200"} 1.0"""
-            in metrics
-        )
-
-
-def test_prometheus_controller_configurations() -> None:
-    config = create_config(
-        exemplars=lambda a: {"trace_id": "1234"},
-    )
-
-    class CustomPrometheusController(PrometheusController):
-        path: str = "/metrics/custom"
-        openmetrics_format: bool = True
-
-    @get("/test")
-    def test() -> dict:
-        return {"hello": "world"}
-
-    with create_test_client([test, CustomPrometheusController], middleware=[config.middleware]) as client:
-        client.get("/test")
-
-        metrics_exporter_response = client.get("/metrics/custom")
-
-        assert metrics_exporter_response.status_code == HTTP_200_OK
-        metrics = metrics_exporter_response.content.decode()
-
-        assert (
-            """litestar_requests_total{app_name="litestar",method="GET",path="/test",status_code="200"} 1.0 # {trace_id="1234"} 1.0"""
-            in metrics
-        )
-
-
-def test_prometheus_with_websocket() -> None:
-    config = create_config()
-
-    @websocket_listener("/test")
-    def test(data: str) -> dict:
-        return {"hello": data}
-
-    with create_test_client([test, PrometheusController], middleware=[config.middleware]) as client:
-        with client.websocket_connect("/test") as websocket:
-            websocket.send_text("litestar")
-            websocket.receive_json()
-
-        metrics_exporter_response = client.get("/metrics")
-
-        assert metrics_exporter_response.status_code == HTTP_200_OK
-        metrics = metrics_exporter_response.content.decode()
-
-        assert (
-            """litestar_requests_total{app_name="litestar",method="websocket",path="/test",status_code="200"} 1.0"""
-            in metrics
-        )
-
-
-@pytest.mark.parametrize("env_var", ["PROMETHEUS_MULTIPROC_DIR", "prometheus_multiproc_dir"])
-def test_procdir(monkeypatch: MonkeyPatch, tmp_path: Path, mocker: MockerFixture, env_var: str) -> None:
-    proc_dir = tmp_path / "something"
-    proc_dir.mkdir()
-    monkeypatch.setenv(env_var, str(proc_dir))
-    config = create_config()
-    mock_registry = mocker.patch("litestar.contrib.prometheus.controller.CollectorRegistry")
-    mock_collector = mocker.patch("litestar.contrib.prometheus.controller.multiprocess.MultiProcessCollector")
 
-    with create_test_client([PrometheusController], middleware=[config.middleware]) as client:
-        client.get("/metrics")
 
-    mock_collector.assert_called_once_with(mock_registry.return_value)
+def purge_module(module_names: list[str], path: str | Path) -> None:
```

Comment:
can we move this to `conftest.py` as a fixture? I noticed we seem to be using this for all of these "contrib to plugin" PRs.

Reply:
Maybe we can do this in a followup PR, and move all of the references to it?